File: Data Structures & Algorithms/search-for-word/submission-1.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:
        ROWS, COLS = len(board), len(board[0])
        visited = set()
        
        def dfs(r, c, idx):
            if idx == len(word):
                return True
            if r >= ROWS or c >= COLS or min(r, c) < 0 or board[r][c] != word[idx] or (r, c) in visited:
                return False
            visited.add((r, c))
            res = dfs(r+1, c, idx+1) or dfs(r, c+1, idx+1) or dfs(r-1, c, idx+1) or dfs(r, c-1, idx+1)
            visited.remove((r, c))
            return res

        for r in range(ROWS):
            for c in range(COLS):
                if dfs(r,c,0):
                    return True
        return False 

File: Data Structures & Algorithms/reverse-bits/submission-1.java
class Solution {
    public int reverseBits(int n) {
        if(n == 0) return 0;
        int res = 0;
        for(int i = 0; i < 32; ++i) {
            if((n >> i & 1) == 1) {
                res |= (1 << (31 - i));
            }
        }
        return res;
    }
}
